### DISCUSSION TITLE
Routes without index files

### DISCUSSION THREAD

**Discussion by special-character** [vercel/next.js#14280] [Help] [answered]
I took out index files I have for non-supported routes but now those routes show the file structure instead of using the 404 page when I visit them.

<img width="237" alt="Screen Shot 2020-06-17 at 10 43 22" src="https://user-images.githubusercontent.com/5407471/84931399-9b40a500-b087-11ea-983e-a05c3caefe6d.png">

ex structure:

`pages/log/today` -> valid
`pages/log` -> invalid

I took out `pages/log` but it doesn't show the 404 page when I visit `/log`. Does anyone know why it shows the folder structure?

**Comment by desaijay315** [comment]
Share screenshot of your structure please

**Reply by special-character** [reply]
This was the original structure:

<img width="159" alt="Screen Shot 2020-06-17 at 11 07 57" src="https://user-images.githubusercontent.com/5407471/84933504-d690a300-b08a-11ea-883d-0a8ee4a6cad5.png">


I removed the index.tsx file directly under the log folder. So `/log` now effectively points to nothing in the file structure. I thought that would cause a 404 but I seem to be wrong.

**Comment by jamesmosier** [comment, marked as answer]
Is this happening locally? Or on your server? If it is on your server, what hosting are you using?

I tested out this locally and going to `localhost:3000/log` results in a 404. 

<img width="160" alt="Screen Shot 2020-06-17 at 8 17 41 PM" src="https://user-images.githubusercontent.com/2854919/84963525-a02d4500-b0d7-11ea-9d83-12930aeaa979.png">

**Reply by special-character** [reply]
@jamesmosier This is just happening on the server and not locally. I bet you are correct. This might be related to deploying with Vercel.

**Reply by jamesmosier** [reply]
Yeah the UI you posted looks like it is from [`serve`](https://www.npmjs.com/package/serve) which was created by Vercel so it is likely just something that is happening there were Vercel is catching all unknown routes and redirecting them to the "index" of your server. I'm not sure what settings exist to prevent this though.

**Reply by special-character** [reply]
Thanks for the help @jamesmosier, I think the fix will probably be in some routing/config/settings with Vercel.
